Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -5240,3 +5241,59 @@ void DuplicateProcessTest::SetUp()
     m_sharedConnection = m_assetProcessorManager->m_stateData.get();
     ASSERT_TRUE(m_sharedConnection);
 }
+
+void MetadataFileTest::SetUp()
+{
+    AssetProcessorManagerTest::SetUp();
+    m_config->AddMetaDataType("foo", "txt");
+}
+
+TEST_F(MetadataFileTest, MetadataFile_SourceFileExtensionDifferentCase)
+{
+
+    using namespace AzToolsFramework::AssetSystem;
+    using namespace AssetProcessor;
+
+    QDir tempPath(m_tempDir.path());
+
+    QString relFileName("Dummy.TXT");
+    QString absPath(tempPath.absoluteFilePath("subfolder1/Dummy.TXT"));
+    QString watchFolder = tempPath.absoluteFilePath("subfolder1");
+    UnitTestUtils::CreateDummyFile(absPath, "dummy");
+
+    JobEntry entry;
+    entry.m_watchFolderPath = watchFolder;
+    entry.m_databaseSourceName = entry.m_pathRelativeToWatchFolder = relFileName;
+    entry.m_jobKey = "txt";
+    entry.m_platformInfo = { "pc", {"host", "renderer", "desktop"} };
+    entry.m_jobRunKey = 1;
+
+    QString productPath(m_normalizedCacheRootDir.absoluteFilePath("outputfile.TXT"));
+    UnitTestUtils::CreateDummyFile(productPath);
+
+    AssetBuilderSDK::ProcessJobResponse jobResponse;
+    jobResponse.m_resultCode = AssetBuilderSDK::ProcessJobResult_Success;
+    jobResponse.m_outputProducts.push_back(AssetBuilderSDK::JobProduct(productPath.toUtf8().data()));
+
+    QMetaObject::invokeMethod(m_assetProcessorManager.get(), "AssetProcessed", Qt::QueuedConnection, Q_ARG(JobEntry, entry), Q_ARG(AssetBuilderSDK::ProcessJobResponse, jobResponse));
+
+    ASSERT_TRUE(BlockUntilIdle(5000));
+
+    // Creating a metadata file for the source assets
+    // APM should process the source asset if a metadafile is detected
+    // We are intentionally having a source file with a different file extension casing than the one specified in the metadata rule.
+    QString metadataFile(tempPath.absoluteFilePath("subfolder1/Dummy.foo"));
+    UnitTestUtils::CreateDummyFile(metadataFile, "dummy");
+
+    // Capture the job details as the APM inspects the file.
+    JobDetails jobDetails;
+    auto connection = QObject::connect(m_assetProcessorManager.get(), &AssetProcessorManager::AssetToProcess, [&jobDetails](JobDetails job)
+        {
+            jobDetails = job;
+        });
+
+    m_assetProcessorManager->AssessAddedFile(tempPath.absoluteFilePath(metadataFile));
+
+    ASSERT_TRUE(BlockUntilIdle(50000));
```

Comment:
This is 10x as long as the previous block until idle. If this test is stuck this long, wouldn't that mean there's a somewhere in the asset processor? Maybe this should be a much shorter time.

Reply:
it should be 5000 and not 50000, will fix

Reply:
fixed, actually i had increased it for testing and forgot to revert back.